diff -rupN cmus-20251212.old/Makefile cmus-20251212/Makefile
--- cmus-20251212.old/Makefile	2025-12-11 23:27:42.000000000 +0100
+++ cmus-20251212/Makefile	2026-08-14 04:38:41.033755287 +0200
@@ -255,9 +255,11 @@ $(man7): Doc/ttman
 %.7: %.txt
 	$(call cmd,ttman)
 
+Doc/ttman.o: HOST_CFLAGS += -fPIC
 Doc/ttman.o: Doc/ttman.c
 	$(call cmd,hostcc,)
 
+Doc/ttman: HOST_LDFLAGS += -fPIC
 Doc/ttman: Doc/ttman.o
 	$(call cmd,hostld,)
 
diff -rupN cmus-20251212.old/scripts/checks.sh cmus-20251212/scripts/checks.sh
--- cmus-20251212.old/scripts/checks.sh	2025-12-11 23:27:42.000000000 +0100
+++ cmus-20251212/scripts/checks.sh	2026-08-14 04:38:41.033808858 +0200
@@ -185,8 +185,8 @@ check_host_cc()
 {
 	var_default HOSTCC gcc
 	var_default HOSTLD $HOSTCC
-	var_default HOST_CFLAGS "-g -O2 -Wall"
-	var_default HOST_LDFLAGS ""
+	var_default HOST_CFLAGS "-g -O2 -Wall -fPIC"
+	var_default HOST_LDFLAGS "-fPIC"
 	check_program $HOSTCC || return 1
 	makefile_vars HOSTCC HOSTLD HOST_CFLAGS HOST_LDFLAGS
 	__check_lang=c
