diff -rupN cmus-2.12.0.old/op/aaudio.c cmus-2.12.0/op/aaudio.c
--- cmus-2.12.0.old/op/aaudio.c	2024-10-26 12:27:42.000000000 +0200
+++ cmus-2.12.0/op/aaudio.c	2025-08-23 09:01:11.165067649 +0200
@@ -555,7 +555,10 @@ static int op_aaudio_open(sample_format_
 	if (API_AT_LEAST(29)) AAudioStreamBuilder_setAllowedCapturePolicy(bld, op_aaudio_opt_allowed_capture);
 	if (API_AT_LEAST(31)) AAudioStreamBuilder_setAttributionTag(bld, "cmus");
 	if (API_AT_LEAST(32)) AAudioStreamBuilder_setSpatializationBehavior(bld, op_aaudio_opt_disable_spatialization ? AAUDIO_SPATIALIZATION_BEHAVIOR_NEVER : AAUDIO_SPATIALIZATION_BEHAVIOR_AUTO);
-
+	
+	// ensure the buffer holds at least 80ms of audio
+	AAudioStreamBuilder_setBufferCapacityInFrames(bld, sf_get_rate(sf) / (1000 / 80));
+	
 	// configure the sample format and channel map
 	rc = configure_aaudio_sf(bld, sf, channel_map, &op.remap);
 	if (rc) {
