diff -rupN cmus-20251007.old/Makefile cmus-20251007/Makefile
--- cmus-20251007.old/Makefile	2025-09-27 16:34:56.099750300 +0200
+++ cmus-20251007/Makefile	2025-10-18 17:34:19.572606331 +0200
@@ -275,17 +275,24 @@ plugins: $(ip-y) $(op-y)
 man: $(man1) $(man7)
 
 install-main: main
-	$(INSTALL) -m755 $(bindir) cmus cmus-remote
+	mkdir -p $(DESTDIR)$(bindir)
+	$(INSTALL) -m755 cmus cmus-remote $(DESTDIR)$(bindir)
 
 install-plugins: plugins
-	$(INSTALL) -m755 $(libdir)/cmus/ip $(ip-y)
-	$(INSTALL) -m755 $(libdir)/cmus/op $(op-y)
+	mkdir -p $(DESTDIR)$(libdir)/cmus/ip
+	mkdir -p $(DESTDIR)$(libdir)/cmus/op
+	$(INSTALL) -m755 $(ip-y) $(DESTDIR)$(libdir)/cmus/ip
+	$(INSTALL) -m755 $(op-y) $(DESTDIR)$(libdir)/cmus/op
 
 install-data: man
-	$(INSTALL) -m644 $(datadir)/cmus $(data)
-	$(INSTALL) -m644 $(mandir)/man1 $(man1)
-	$(INSTALL) -m644 $(mandir)/man7 $(man7)
-	$(INSTALL) -m755 $(exampledir) cmus-status-display
+	mkdir -p $(DESTDIR)$(datadir)/cmus
+	mkdir -p $(DESTDIR)$(mandir)/man1
+	mkdir -p $(DESTDIR)$(mandir)/man7
+	mkdir -p $(DESTDIR)$(exampledir)
+	$(INSTALL) -m644 $(data) $(DESTDIR)$(datadir)/cmus
+	$(INSTALL) -m644 $(man1) $(DESTDIR)$(mandir)/man1
+	$(INSTALL) -m644 $(man7) $(DESTDIR)$(mandir)/man7
+	$(INSTALL) -m755 cmus-status-display $(DESTDIR)$(exampledir)
 
 install: all install-main install-plugins install-data
 
