diff -rupN cmus-2.12.0.old/player.c cmus-2.12.0/player.c
--- cmus-2.12.0.old/player.c	2025-08-23 10:03:12.612412827 +0200
+++ cmus-2.12.0/player.c	2025-08-23 10:06:42.068353637 +0200
@@ -31,6 +31,7 @@
 #include "cmus.h"
 #include "lib.h"
 #include "pl_env.h"
+#include "ui_curses.h"
 
 #include <stdio.h>
 #include <stdlib.h>
@@ -1329,12 +1330,12 @@ void player_set_op(const char *name)
 		rc = op_select(name);
 	}
 	
-	/*
-	 * if plugin is null we either never selected one and will now try any,
-	 * or we tried and failed and will now try to fallback to any
-	 */
-	if (op_get_current() == NULL) {
+	/* when at startup and plugin is null, op_select_any() */
+	if (!ui_initialized && op_get_current() == NULL) {
 		if (rc)
+			/*
+			 * error if we are falling back because the specified init plugin failed
+			 */
 			player_op_error(rc, "selecting output plugin '%s'", name);
 
 		d_print("selecting first initialized op\n");
