diff -rupN cmus-2.12.0.old/track_info.c cmus-2.12.0/track_info.c
--- cmus-2.12.0.old/track_info.c	2024-10-26 12:27:42.000000000 +0200
+++ cmus-2.12.0/track_info.c	2025-08-23 09:45:10.544821317 +0200
@@ -254,6 +254,7 @@ int track_info_cmp(const struct track_in
 		case SORT_ORIGINALDATE:
 		case SORT_PLAY_COUNT:
 		case SORT_BPM:
+		case SORT_DURATION:
 			res = getentry(a, key, int) - getentry(b, key, int);
 			break;
 		case SORT_FILEMTIME:
@@ -293,6 +294,7 @@ static const struct {
 	{ "album",		SORT_ALBUM		},
 	{ "title",		SORT_TITLE		},
 	{ "play_count",		SORT_PLAY_COUNT		},
+	{ "duration",		SORT_DURATION		},
 	{ "tracknumber",	SORT_TRACKNUMBER	},
 	{ "discnumber",		SORT_DISCNUMBER		},
 	{ "totaldiscs",		SORT_TOTALDISCS		},
@@ -316,6 +318,7 @@ static const struct {
 	{ "-album",		REV_SORT_ALBUM		},
 	{ "-title",		REV_SORT_TITLE		},
 	{ "-play_count", 	REV_SORT_PLAY_COUNT	},
+	{ "-duration",		REV_SORT_DURATION	},
 	{ "-tracknumber",	REV_SORT_TRACKNUMBER	},
 	{ "-discnumber",	REV_SORT_DISCNUMBER	},
 	{ "-totaldiscs",	REV_SORT_TOTALDISCS	},
diff -rupN cmus-2.12.0.old/track_info.h cmus-2.12.0/track_info.h
--- cmus-2.12.0.old/track_info.h	2024-10-26 12:27:42.000000000 +0200
+++ cmus-2.12.0/track_info.h	2025-08-23 09:46:07.493848380 +0200
@@ -89,6 +89,7 @@ typedef size_t sort_key_t;
 #define SORT_GENRE         	offsetof(struct track_info, collkey_genre)
 #define SORT_COMMENT       	offsetof(struct track_info, collkey_comment)
 #define SORT_ALBUMARTIST   	offsetof(struct track_info, collkey_albumartist)
+#define SORT_DURATION		offsetof(struct track_info, duration)
 #define SORT_PLAY_COUNT   	offsetof(struct track_info, play_count)
 #define SORT_FILENAME      	offsetof(struct track_info, filename)
 #define SORT_FILEMTIME     	offsetof(struct track_info, mtime)
@@ -101,6 +102,7 @@ typedef size_t sort_key_t;
 #define REV_SORT_ARTIST		(REV_SORT__START + offsetof(struct track_info, collkey_artist))
 #define REV_SORT_ALBUM          (REV_SORT__START + offsetof(struct track_info, collkey_album))
 #define REV_SORT_TITLE          (REV_SORT__START + offsetof(struct track_info, collkey_title))
+#define REV_SORT_DURATION	(REV_SORT__START + offsetof(struct track_info, duration))
 #define REV_SORT_PLAY_COUNT   	(REV_SORT__START + offsetof(struct track_info, play_count))
 #define REV_SORT_TRACKNUMBER    (REV_SORT__START + offsetof(struct track_info, tracknumber))
 #define REV_SORT_DISCNUMBER     (REV_SORT__START + offsetof(struct track_info, discnumber))
