diff -Nru a/Makefile b/Makefile
--- a/Makefile	2015-07-13 12:00:56.000000000 +0200
+++ b/Makefile	2017-10-11 22:07:01.221564512 +0200
@@ -262,17 +262,24 @@
 man: $(man1) $(man7)
 
 install-main: main
-	$(INSTALL) -m755 $(bindir) cmus cmus-remote
+	mkdir -p $(DESTDIR)$(bindir)
+	$(INSTALL) -m755 cmus cmus-remote $(DESTDIR)$(bindir)
 
 install-plugins: plugins
-	$(INSTALL) -m755 $(libdir)/cmus/ip $(ip-y)
-	$(INSTALL) -m755 $(libdir)/cmus/op $(op-y)
+	mkdir -p $(DESTDIR)$(libdir)/cmus/ip
+	mkdir -p $(DESTDIR)$(libdir)/cmus/op
+	$(INSTALL) -m755 $(ip-y) $(DESTDIR)$(libdir)/cmus/ip
+	$(INSTALL) -m755 $(op-y) $(DESTDIR)$(libdir)/cmus/op
 
 install-data: man
-	$(INSTALL) -m644 $(datadir)/cmus $(data)
-	$(INSTALL) -m644 $(mandir)/man1 $(man1)
-	$(INSTALL) -m644 $(mandir)/man7 $(man7)
-	$(INSTALL) -m755 $(exampledir) cmus-status-display
+	mkdir -p $(DESTDIR)$(datadir)/cmus
+	mkdir -p $(DESTDIR)$(mandir)/man1
+	mkdir -p $(DESTDIR)$(mandir)/man7
+	mkdir -p $(DESTDIR)$(exampledir)
+	$(INSTALL) -m644 $(data) $(DESTDIR)$(datadir)/cmus
+	$(INSTALL) -m644 $(man1) $(DESTDIR)$(mandir)/man1
+	$(INSTALL) -m644 $(man7) $(DESTDIR)$(mandir)/man7
+	$(INSTALL) -m755 cmus-status-display $(DESTDIR)$(exampledir)
 
 install: all install-main install-plugins install-data
 
