diff -rupN cmus-2.12.0.old/configure cmus-2.12.0/configure
--- cmus-2.12.0.old/configure	2024-10-26 12:27:42.000000000 +0200
+++ cmus-2.12.0/configure	2025-08-23 09:37:46.601482252 +0200
@@ -471,11 +471,12 @@ check_aac()
 
 check_ffmpeg()
 {
-	pkg_config FFMPEG "libavformat libavcodec libswresample" || return $?
+	pkg_config FFMPEG "libavformat libavcodec libswresample libavutil" || return $?
 	# check the existence of specific headers since they've been renamed before
 	check_header "libavformat/avformat.h" $FFMPEG_CFLAGS || return $?
 	check_header "libavcodec/avcodec.h" $FFMPEG_CFLAGS || return $?
 	check_header "libswresample/swresample.h" $FFMPEG_CFLAGS || return $?
+	check_header "libavutil/avutil.h" $FFMPEG_CFLAGS || return $?
 	# ffmpeg api changes so frequently that it is best to compile the module
 	libs="$LDDLFLAGS $FFMPEG_LIBS"
 	cflags="$SOFLAGS $FFMPEG_CFLAGS"
